### PR TITLE
CAPETOWN - SIMBA PFAIRA - JAVASCRIPT - CORE 1

### PR DESCRIPTION
**Pull request by sirpfaira**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

File: exercises/F-strings-methods/exercise.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
 // Start by creating a variable `message`
-
+let myName = "Sam";
+let message =
+  "My name is " +
+  myName +
+  " and my name is " +
+  myName.length +
+  " characters long";
```
Comment (mulindierick) [inline exercises/F-strings-methods/exercise.js:8]: could there be a reason you didn't do the concatenation in one line?

File: exercises/L-functions-nested/exercise.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+function getPercentage(num1, num2) {
+  return Math.round((num1 / (num1 + num2)) * 100);
+}
+
+function createMessage(kind, percent) {
+  console.log(`Percentage ${kind}: ${percent}%`);
+}
+
+createMessage("students", getPercentage(numberOfStudents, numberOfMentors));
+createMessage("mentors", getPercentage(numberOfMentors, numberOfStudents));
```
Comment (mulindierick) [inline exercises/L-functions-nested/exercise.js:13]: this was smart. I had to create three functions for now own and it was hectic

File: mandatory/3-function-output.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+
+//This method append "w2" to "w1" and return the new combined string
 function s(w1, w2) {
   return w1.concat(w2);
 }
 
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  return `${firstWord} ${secondWord} ${thirdWord}`;
```
Comment (mulindierick) [inline mandatory/3-function-output.js:18]: I thought we were supposed to use the concat method

**Review by mulindierick — COMMENTED** [review]
I love your coding style. it's clean and easily readable
well done

File: mandatory/1-syntax-errors.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-    return a + b + c;
+function addNumbers(a, b, c) {
+  return a + b + c;
 }
 
-function introduceMe(name, age)
-return "Hello, my name is " + name "and I am " age + "years old";
-
+function introduceMe(name, age) {
+  return "Hello, my name is " + name + " and I am " + age + " years old";
```
Comment (Ntshembo-Hlongwane1) [inline mandatory/1-syntax-errors.js:8]: Hey great work just some code refactoring tip here...

```diff
- return "Hello, my name is " + name + " and I am " + age + " years old";
+ return `Hello my name is ${name} and I am ${age} years old`;
```

- So your solution there works but prone to errors that can be caused **by mistake you can miss on spacing your quotation marks then the words get attached together as one word which is something you did not intend to do** 
- So template literals tend to be the **most effective in concatenating strings etc**

Comment (sirpfaira) [reply mandatory/1-syntax-errors.js:8]: Thanks i appreciate

**Pull request closed without merging**